cpu/cc2538/periph/spi.c

```diff
@@ -69,33 +89,18 @@ void spi_init(spi_t bus)
 
 void spi_init_pins(spi_t bus)
 {
-    switch ((uintptr_t)spi_config[bus].dev) {
-        case (uintptr_t)SSI0:
-            IOC_PXX_SEL[spi_config[bus].mosi_pin] = SSI0_TXD;
-            IOC_PXX_SEL[spi_config[bus].sck_pin ] = SSI0_CLK_OUT;
-            IOC_PXX_SEL[spi_config[bus].cs_pin  ] = SSI0_FSS_OUT;
-
-            IOC_SSIRXD_SSI0 = spi_config[bus].miso_pin;
-            break;
-
-        case (uintptr_t)SSI1:
-            IOC_PXX_SEL[spi_config[bus].mosi_pin] = SSI1_TXD;
-            IOC_PXX_SEL[spi_config[bus].sck_pin ] = SSI1_CLK_OUT;
-            IOC_PXX_SEL[spi_config[bus].cs_pin  ] = SSI1_FSS_OUT;
-
-            IOC_SSIRXD_SSI1 = spi_config[bus].miso_pin;
-            break;
-    }
+    assert(bus < SPI_NUMOF);
 
-    IOC_PXX_OVER[spi_config[bus].mosi_pin] = IOC_OVERRIDE_OE;
-    IOC_PXX_OVER[spi_config[bus].miso_pin] = IOC_OVERRIDE_DIS;
-    IOC_PXX_OVER[spi_config[bus].sck_pin ] = IOC_OVERRIDE_OE;
-    IOC_PXX_OVER[spi_config[bus].cs_pin  ] = IOC_OVERRIDE_OE;
-
-    gpio_hardware_control(spi_config[bus].mosi_pin);
-    gpio_hardware_control(spi_config[bus].miso_pin);
-    gpio_hardware_control(spi_config[bus].sck_pin);
-    gpio_hardware_control(spi_config[bus].cs_pin);
+    if (SPI(bus) == 0) {
```

Comment:
This if clause can be removed. The ```SSI1_*``` values are ```SSI0_*``` + 4, thus using e.g., ```SSI0_TXD + (4*bus)``` should work.

Reply:
there are only 2 SPI devices, so IMHO the `if ... else` here is _more_ readable than `SSI0_TXD + (4*bus)`. Besides it omits the multiplication at runtime (though I'm not sure if a compiler can optimise that somehow anyway).